src/common/TrackedOp.h

```diff
@@ -179,11 +179,14 @@ class TrackedOp {
   const utime_t& get_initiated() const {
     return initiated_at;
   }
-  // This function maybe needs some work; assumes last event is completion time
+
   double get_duration() const {
-    return events.empty() ?
-      0.0 :
-      (events.rbegin()->first - get_initiated());
+    if (events.empty())
+      return 0.0;
```

Comment:
I think we still need to calculate the time since it was initiated even if it hasn't made any progress since then, rather than just returning 0 if it's not hit a marked event point.